recipe/meta.yaml

```diff
@@ -28,11 +28,11 @@ source:
 
 build:
   number: 0
-  skip: true  # [osx or win]
+  skip: true  # [osx]
 
 outputs:
   - name: cuda-nvcc-tools
-    files:
+    files:                                  # [linux]
```

Comment:
Since `files` is effectively empty on Linux, ran into a build issue [on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=764279&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=f5d15007-a01c-5ad8-c9ce-4d519d3b275f&l=372 ) (also here is [the attached log]( https://github.com/conda-forge/cuda-nvcc-impl-feedstock/files/12373908/cuda-nvcc-impl-feedstock_20230817.4_win_64.txt )). Here is the relevant traceback:

```python
Traceback (most recent call last):
  File "C:\Miniforge\Scripts\conda-mambabuild-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 256, in main
    call_conda_build(action, config)
  File "C:\Miniforge\lib\site-packages\boa\cli\mambabuild.py", line 228, in call_conda_build
    result = api.build(
  File "C:\Miniforge\lib\site-packages\conda_build\api.py", line 253, in build
    return build_tree(
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 3799, in build_tree
    packages_from_this = build(
  File "C:\Miniforge\lib\site-packages\conda_build\build.py", line 2852, in build
    for f in output_d.get("files", []):
TypeError: 'NoneType' object is not iterable
```

Basically `conda-build` doesn't see an empty list `[]`, but instead sees `None`, which can't be iterated over (hence the error)

Filed this with a minimal reproducer in upstream issue ( https://github.com/conda/conda-build/issues/4972 )

For now just adding a selector here to ignore `files` when empty (like on Windows) was a simple way to workaround the issue. We would just want to remember to remove this whenever Windows packages need files added here